packages/circuit-ui/components/ImageInput/ImageInput.docs.mdx

```diff
@@ -0,0 +1,45 @@
+import { Status, Props, Story } from '../../../../.storybook/components';
+
+# ImageInput
+
+<Status.Stable />
+
+The ImageInput component allows users to upload images. It can be used on its own or as part of a form.
+
+<Story id="forms-imageinput--base" />
+
+<Props />
+
+## Usage guidelines
+
+Refer to the `Stateful` component story for a simplified code usage example.
+
+- **Do** use the onChange callback to immediately trigger an image upload when the user has selected one, even when the component is being used as part of a form (it will make form submission faster).
+- **Do not** use the ImageInput component if the image is read-only for the user. Instead, use the visual component from the ImageInput's `component` prop by itself, for example the [Avatar](Components/Avatar).
+
+## Input states
+
+### Existing image
+
+When an `src` prop is provided, the ImageInput renders a "Clear" icon instead of the "Add" icon. It also forwards the prop to the visual component passed as a prop. Clicking the "Clear" icon clears the input.
+
+<Story id="forms-imageinput--with-image" />
+
+### Invalid
+
+When an image upload fails, set the `invalid` prop to true to render invalid styles, and show the user a clear and concise error message using the `validationHint` prop.
+
+<Story id="forms-imageinput--invalid" />
+
+## Custom component
+
+The ImageInput has drop-in support for the [Avatar](Components/Avatar) component as a visual element.
+
+However, where the end result image needs to be in a landscape view, a custom component with different dimensions can be used to reflect the final image's aspect ratio.
+
+There are some caveats with this approach:
+
+- A custom placeholder will also need to be provided to match the custom aspect ratio. Refer to the `CustomComponent` story for a code example.
+- Currently, the `ImageInput` only supports rectangular components with a fixed border-radius, in order to correctly apply state-specific styles.
```

Comment:
Does this mean the visual component needs to have a specific border radius? I thought these styles were applied by the ImageInput?

Reply:
Yes 😕 this is because of the state-specific styles like the focus or hover styles. They're applied by the ImageInput in pseudo-elements, which need to be given the same border-radius as the visual component.

This would be fixed by using `filter: brightness()` instead of pseudo-elements, but other than this I haven't thought of other solutions with good browser support

Reply:
C.f. for example this occurrence in the overlay styles (which handle the different shades of grey overlays for the hover/active states):

https://github.com/sumup-oss/circuit-ui/pull/922/files#diff-6d71d84fd0ca418fcb16199c540e16e8f7a35ffbbaea54fa884d0dbf56320502R149

Reply:
Ah, so the visual component can have a smaller/no border radius (in that case it'll be clipped) but not a larger one (because the overlay will overflow)?

Reply:
Yes 👍